c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -2965,49 +3041,35 @@ static bool retrieve_value_for_header_from_handle(JSContext *cx, HandleObject se
   MOZ_ASSERT(mode != Mode::Standalone);
   uint32_t handle = detail::handle(self);
 
-  size_t name_len;
+  xqd_world_string_t str;
   RootedString name_str(cx, name.toString());
-  UniqueChars name_chars = encode(cx, name_str, &name_len);
-
-  RootedString val_str(cx);
-  OwnedHostCallBuffer buffer;
-
-  MULTI_VALUE_HOSTCALL(
-      {
-        FastlyStatus result;
-        if (mode == Headers::Mode::ProxyToRequest) {
-          RequestHandle request = {handle};
-          result = convert_to_fastly_status(
-              xqd_req_header_values_get(request, name_chars.get(), name_len, buffer.get(),
-                                        HEADER_MAX_LEN, cursor, &ending_cursor, &nwritten));
-        } else {
-          ResponseHandle response = {handle};
-          result = convert_to_fastly_status(
-              xqd_resp_header_values_get(response, name_chars.get(), name_len, buffer.get(),
-                                         HEADER_MAX_LEN, cursor, &ending_cursor, &nwritten));
-        }
+  UniqueChars name_chars = encode(cx, name_str, &str.len);
+  str.ptr = name_chars.get();
 
-        if (!HANDLE_RESULT(cx, result))
-          return false;
-      },
-      {
-        size_t offset = 0;
-        for (size_t i = 0; i < nwritten; i++) {
-          if (buffer.get()[i] == '\0') {
-            val_str = JS_NewStringCopyUTF8N(cx, JS::UTF8Chars(buffer.get() + offset, i - offset));
-            if (!val_str)
-              return false;
+  fastly_option_list_string_t ret;
 
-            value.setString(val_str);
+  FastlyStatus result;
+  if (mode == Headers::Mode::ProxyToRequest) {
+    result = convert_to_fastly_status(xqd_fastly_http_req_header_values_get(handle, &str, &ret));
+  } else {
+    result = convert_to_fastly_status(xqd_fastly_http_resp_header_values_get(handle, &str, &ret));
+  }
 
-            if (!append_header_value_to_map(cx, self, name, value))
-              return false;
+  if (!ret.is_some)
```

Comment:
Do we need to check `result` here?
```suggestion
  if (!HANDLE_RESULT(cx, result)){
    return false;
  }
  if (!ret.is_some)
```

Reply:
Yes, it seems the clang upgrade caught this bug!